backend/src/modules/tournament-tree/core/domain/TournamentTree.ts

```diff
@@ -19,6 +19,18 @@ export class TournamentTree {
     this.tournamentId = props.tournamentId;
   }
 
+  private finishMatchesFromLevelZeroWhereIsOnlyOneTeam(): void {
+    this.getTournamentTreeArray().map((match) => {
+      const isMatchFromLevelZeroWithoutOneTeam = match.fightingTeamsGroupLevel === 0 && (!match.firstTeam || !match.secondTeam);
+      if (isMatchFromLevelZeroWithoutOneTeam && match.firstTeam) {
+        this.finishMatch(match.fightingTeamsGroupId.raw, match.firstTeam.teamId.raw);
+      }
+      if (isMatchFromLevelZeroWithoutOneTeam && match.secondTeam) {
+        this.finishMatch(match.fightingTeamsGroupId.raw, match.secondTeam.teamId.raw);
+      }
+    });
+  }
```

Comment:
Jak masz ifa w jakiejś operacji na tablicach to bardzo możliwe, że jest to miejsce na filter :) 
Proponuję:
```suggestion
    this.tournamentTreeArray
      .filter(match => match.fightingTeamsGroupLevel === 0 && (!match.firstTeam || !match.secondTeam))
      .map((match) => {
      if (match.firstTeam) {
        this.finishMatch(match.fightingTeamsGroupId.raw, match.firstTeam.teamId.raw);
      }
      if (match.secondTeam) {
        this.finishMatch(match.fightingTeamsGroupId.raw, match.secondTeam.teamId.raw);
      }
    });
```

Reply:
Jeśli chodzi o część związaną z kopią to jasne rozumiem, moja nieuwaga - poprawione.

Reply:
Nie sądziłem, ze nie powinniśmy korzystać z metod publicznych wewnątrz klasy, ale wydaje się, ze ma to sens, tak jak piszesz. Postaram się pamiętać na przyszłość